src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentSelector.cs

```diff
@@ -3,44 +3,41 @@
 
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Linq;
-using System.Reflection;
 using Microsoft.AspNet.Mvc.Core;
 using Microsoft.Framework.Internal;
 
-namespace Microsoft.AspNet.Mvc
+namespace Microsoft.AspNet.Mvc.ViewComponents
 {
+    /// <summary>
+    /// Default implementation of <see cref="IViewComponentSelector"/>.
+    /// </summary>
     public class DefaultViewComponentSelector : IViewComponentSelector
```

Comment:
nope. This is all consistent with `IActionSelector` on purpose. The fact that a descriptor exists doesn't change the logical operation we're performing.